iceoryx_posh/include/iceoryx_posh/internal/popo/listener.inl

```diff
@@ -61,31 +67,157 @@ inline cxx::expected<ListenerError> Listener::attachEvent(
         .and_then([&](auto& eventId) {
             NotificationAttorney::enableEvent(
                 eventOrigin,
-                TriggerHandle(*m_conditionVariableData, {*this, &Listener::removeTrigger}, eventId),
+                TriggerHandle(*m_conditionVariableData, {*this, &ListenerImpl<Capacity>::removeTrigger}, eventId),
                 eventType);
         });
 }
 
+template <uint64_t Capacity>
 template <typename T, typename EventType, typename>
-inline void Listener::detachEvent(T& eventOrigin, const EventType eventType) noexcept
+inline void ListenerImpl<Capacity>::detachEvent(T& eventOrigin, const EventType eventType) noexcept
 {
     static_assert(IS_EVENT_ENUM<EventType>,
                   "Only enums with an underlying EventEnumIdentifier can be attached/detached to the Listener");
     NotificationAttorney::disableEvent(eventOrigin, eventType);
 }
 
+template <uint64_t Capacity>
 template <typename T>
-inline void Listener::detachEvent(T& eventOrigin) noexcept
+inline void ListenerImpl<Capacity>::detachEvent(T& eventOrigin) noexcept
 {
     NotificationAttorney::disableEvent(eventOrigin);
 }
 
-inline constexpr uint64_t Listener::capacity() noexcept
+template <uint64_t Capacity>
```

Comment:
The code in listener.inl was only moved without change from listener.cpp. This was required since the listener is now a template class.